### ISSUE TITLE
/* jshint -W101 */ not working when withing IIFE

### ISSUE THREAD

**Issue by batista** [jshint/jshint#3017]
~~I believe there's a regression between versions 2.9.2 and 2.9.3.~~

The following ~~started throwing a warning since 2.9.3~~ throws an error `line 5, col 133, Line is too long. (W101)`:

``` javascript
(function() {
  function myFunction(myVar) {
    /* jshint -W101 */
    if (myVar) {
      var longVariableContent = 'loooooooooooooooooooooooooooooooooooooooooooooooooong liiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiine';
    }
  }
})();
```

note that without the IIFE the warning doesn't trigger, the following passes the test:

``` javascript
  function myFunction(myVar) {
    /* jshint -W101 */
    if (myVar) {
      var longVariableContent = 'loooooooooooooooooooooooooooooooooooooooooooooooooong liiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiine';
    }
  }
```

EDIT: I realized that it has nothing to do with a recent update, although in my real use case, the warning has only started triggering after 2.9.3. I can't reproduce yet in a test though.

**Comment by jugglinmike** [comment]
Thanks for the report! One detail missing from your test case is the `maxlen`
option. There are also a few details which aren't strictly necessary, so I've
cleaned things up a bit:

```js
// jshint maxlen: 25
(function() {
  /* jshint -W101 */
  return "ooooooooooooooo";
})();
```

This is actually another case of a recently-reported bug; see gh-3013 for way
too much detail on what is going wrong here (along with an explanation as to
why it appears as a version 2.9.3 regression in your application).

I'm especially glad you reported the problem because it demonstrates how my
patch at gh-3016 is incomplete.

**Comment by batista** [comment]
fixed with merge of #3016

**Comment by jugglinmike** [comment]
Thanks for following up on this, Sérgio!
